mxcubeweb/core/server/resource_handler.py

```diff
@@ -346,7 +346,7 @@ def _validate_param_data(
                 # We only handle a single argument when using Pydantic models
                 # We handle complex structures by checking if param_data
                 # contains a value or a dictionary.
-                if isinstance(param_data[param_name], dict):
+                if isinstance(param_data.get(param_name), dict):
```

Comment:
Hm, not sure I see what the difference is, are you sure this solves the issue ?

Reply:
Ok, now I see. so with `.get` you avoid the `KeyError` and you don't get the issue. So that would fix the problem (or avoid it). However, we can remove the entire if statement. It was added as a convince/guard for when several dictionaries are passed (so that only the first one is used). But its up to us to make that it does not happen

```
                # We only handle a single argument when using Pydantic models
                # We handle complex structures by checking if param_data
                # contains a value or a dictionary.
                if isinstance(param_data[param_name], dict):
                    return param_type.parse_obj(param_data[param_name])
```

Reply:
Hm, actually @elmjag ... wait. Your fix is the good one. However I agree that the comment is a bit confusing. Ill try to explain

Reply:
So my comment above is correct but there is a special case for the `set_value` of an adapter, that can either take N arguments (of native python types) or a PyDantic model. So this `if` statement takes care of the case when we receive a dictionary containing a pydantic model. ie `{value: { .... data ...}}` instead if simply `{....data ...}` which is the case for `set_value`.